src/api.js

```diff
@@ -1,14 +1,13 @@
 'use strict';
-import { constructObj, getResults } from './utilities';
 
-export default (request, searchType) => {
-  const resultsPromise = getResults(request);
+export default (request, searchType, resultsHandler) => {
```

Comment:
Let's deconstruct `resultsHandler` in the function parameters:

```
export default (request, searchType, { processor, objectConstructor }) => {
```

Then you can just use `processor` and `objectConstructor` without `resultsHandler.processor` or `resultsHandler.objectConstructor`

Reply:
Done.